File: Tema 5 - Arrrays/recodificado/ejercicio3/archives/automaticamente/automaticamente.py
"""
Este módulo muestra un MENSAJE incrementando el valor de i.
"""
import sys
import json

if len(sys.argv) > 1:
    NUMERO = json.loads(sys.argv[1])
    MENSAJE = ""

    MAX = max(NUMERO)
    MIN = min(NUMERO)

    MENSAJE += "<table border='1'><tr><th>Indice</th><th>Numero</th></tr>"

    for contador, valor_numero in enumerate(NUMERO):
        DESCRIPCION = valor_numero
        if MAX == MIN:
            DESCRIPCION = f"{valor_numero} max y min"
        elif valor_numero == MAX:
            DESCRIPCION = f"{valor_numero} maximo"
        elif valor_numero == MIN:
            DESCRIPCION = f"{valor_numero} minimo"

        MENSAJE += f"<tr><td>{contador}</td><td>{DESCRIPCION}</td></tr>"

    MENSAJE += "</table>"

    print(f"{MENSAJE}")

File: Tema 5 - Arrrays/recodificado/ejercicio3/archives/automaticamente/automaticamente.php
<?php
if (isset($_GET["numerosAuto"])) {
  $numero = json_decode($_GET["numerosAuto"]);
  $mensaje = "<table border='1'><tr><th>Indice</th><th>Numero</th></tr>";

  $max = max($numero);
  $min = min($numero);

  foreach ($numero as $indice => $elemento) {
    $descripcion = $elemento;
    if ($elemento == $min && $elemento == $max) {
      $descripcion = "$elemento es max y min";
    } elseif ($elemento == $min) {
      $descripcion = "$elemento mínimo";
    } elseif ($elemento == $max) {
      $descripcion = "$elemento máximo";
    } 
    $mensaje .= "<tr><td>$indice</td><td>$descripcion</td></tr>";
  }

  $mensaje .= "</table>";
  echo $mensaje;
}
